src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -93,27 +93,39 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 	}
 	
 	@Override
-	public Collection<AddOnInfoSummary> search(AddOnType type, String query) throws IOException {
+	public Collection<AddOnInfoSummary> search(AddOnType type, String query, String tag) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
-			//Gets all modules of the type specified by variable type
+			//Exact match on type
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
+		if (tag != null) {
+			
+		SearchResult result = client.execute(new Search.Builder(new SearchSourceBuilder()
+				.size(SEARCH_SIZE)
+				.query(QueryBuilders.matchQuery("tags", tag)).toString())
+				.addIndex(AddOnInfoAndVersions.ES_INDEX)
+				.build());
+		return result.getHits(AddOnInfoAndVersions.class).stream()
```

Comment:
I mean that the caller might want to combine other parameters along with tag, like:

    index.search(Type.MODULE, null, "form-entry")
    index.search(Type.OWA, "html", "form-entry")

Your current code doesn't support this, but it would be easy to add.